test_submissions/civic/create_civic_therapy_submissions.ipynb

```diff
@@ -0,0 +1,518 @@
+    "\n",
+    "  :param therapeutic: Therapeutic data\n",
+    "  :param tumor_type_normalized_id: Normalized tumor type concept ID from disease-normalizer\n",
+    "  :param study_id: ID of study\n",
+    "  :return: Tuple containing indication and drug label information. Or tuple containing\n",
+    "    ``None`` values if no regulatory approval or drug label information is found\n",
+    "  \"\"\"\n",
+    "  drug_for_ta = None\n",
+    "  reg_approval = None\n",
+    "  therapy_exts = therapeutic[\"extensions\"]\n",
+    "\n",
+    "  for ext in therapy_exts:\n",
+    "    if ext[\"name\"] == \"therapy_normalizer_data\":\n",
+    "      drug_for_ta = ext[\"value\"][\"label\"]\n",
+    "    elif ext[\"name\"] == \"regulatory_approval\":\n",
+    "      reg_approval = ext[\"value\"]\n",
+    "\n",
+    "  if not drug_for_ta:\n",
+    "    logging.debug(\"%s: no drug_for_ta\", study_id)\n",
+    "    return None, None\n",
+    "\n",
+    "  if not reg_approval:\n",
+    "    logging.debug(\"%s: no regulatory_approval\", study_id)\n",
+    "    return None, None\n",
+    "\n",
+    "  within_indication = _is_within_indication(reg_approval, tumor_type_normalized_id)\n",
+    "\n",
+    "  return within_indication, drug_for_ta"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def _get_ind_and_drug_label(study: dict, study_id: str, tumor_type_normalized_id: str) -> tuple[bool, str] | None:\n",
+    "  \"\"\"Get indication and drug label information for therapeutic procedure\n",
+    "\n",
+    "  :param study: GKS study\n",
+    "  :param study_id: ID for ``study``\n",
+    "  :param tumor_type_normalized_id: Normalized tumor type concept ID from disease-normalizer\n",
+    "  :return: Tuple containing indication and drug label if all components found. Otherwise,\n",
+    "    ``None``\n",
+    "  \"\"\"\n",
+    "  therapeutic = study[\"therapeutic\"]\n",
+    "  drug_for_ta = None\n",
+    "  within_indication = None\n",
+    "\n",
+    "  if therapeutic[\"type\"] == \"TherapeuticAgent\":\n",
+    "    within_indication, drug_for_ta = _get_ta_ind_and_drug_label(\n",
+    "      therapeutic, tumor_type_normalized_id, study_id\n",
+    "    )\n",
+    "  elif therapeutic[\"type\"] == \"CombinationTherapy\":\n",
+    "    drug_for_ta_list = []\n",
+    "    within_indication_list = []\n",
+    "\n",
+    "    for comp in therapeutic[\"components\"]:\n",
+    "      _within_indication, _drug_for_ta = _get_ta_ind_and_drug_label(\n",
+    "        comp, tumor_type_normalized_id, study_id\n",
+    "      )\n",
+    "      if not _drug_for_ta:\n",
+    "        return None\n",
+    "\n",
+    "      drug_for_ta_list.append(_drug_for_ta)\n",
+    "      within_indication_list.append(_within_indication)\n",
+    "\n",
+    "    drug_for_ta = \";\".join(drug_for_ta_list)\n",
+    "    within_indication = all(within_indication_list)\n",
```

Comment:
@wesleygoar is this correct? Assuming all drugs have to be within indication for combo therapies

Reply:
This is generally true though not a hard and fast rule.

Reply:
@korikuzma Don't change it. This is the proper way to do it.